subsys/net/ip/icmpv6.c

```diff
@@ -124,8 +129,270 @@ static inline void echo_reply_debug(struct net_pkt *pkt)
 #define echo_reply_debug(pkt)
 #endif /* CONFIG_NET_DEBUG_ICMPV6 */
 
+struct net_buf *net_icmpv6_set_chksum(struct net_pkt *pkt,
+				      struct net_buf *frag)
+{
+	struct net_icmp_hdr *icmp_hdr;
+	u16_t chksum = 0;
+	u16_t pos;
+
+	icmp_hdr = net_pkt_icmp_data(pkt);
+	if (net_icmp_header_fits(pkt, icmp_hdr)) {
+		icmp_hdr->chksum = 0;
+		icmp_hdr->chksum = ~net_calc_chksum_icmpv6(pkt);
+
+		return frag;
+	}
+
+	frag = net_pkt_write(pkt, frag, net_pkt_ip_hdr_len(pkt) +
+			     net_pkt_ipv6_ext_len(pkt) +
+			     1 + 1 /* type + code */,
```

Comment:
I'd say, instead of using such magic constants, would be better to use offsetof() operation.